src/ios/registerNativeModule.js

```diff
@@ -50,6 +54,73 @@ const addProjectToLibraries = (libraries, file) => {
 };
 
 /**
+ * Given folder, it returns an array of all header files
+ * inside it, ignoring node_modules and examples
+ */
+const getHeadersInFolder = (folder) =>
+  glob
+    .sync('**/*.h', {
+      cwd: folder,
+      nodir: true,
+      ignore: GLOB_EXCLUDE_PATTERN,
+    })
+    .map(file => path.join(folder, file));
+
+/**
+ * Given an array of headers it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+const getHeaderSearchPath = (sourceDir, headers) => {
+  const directories = union(
+    headers.map(path.dirname)
+  );
+
+  if (directories.length === 1) {
+    return `"$(SRCROOT)/${path.relative(sourceDir, directories[0])}"`;
+  }
+
+  const directory = directories.reduce((topMostDir, currentDir) => {
+    return currentDir.split(path.sep) < topMostDir.split(path.sep)
+      ? currentDir
+      : topMostDir;
+  });
+
+  return `"$(SRCROOT)/${path.relative(sourceDir, directory)}/**"`;
+};
```

Comment:
This is needed so for code-push it returns `../node-modules/react-native-code-push` (not recursive because all is in top-level folder), for react-native-quick-actions it returns `../node_modules/react-native-quick-actions/RNQuickActions/RNQuickActions` (all in same folder as well), but for library that has multiple folders (just in case), e.g.:
```
- BVLinearGradient.h
- typicalNestedFolder
-- BVLinearGradientManager.h
```
it will return `.../node_modules/react-native-linear-gradient/**`